File: crates/breez-sdk/bindings/examples/cli/langs/swift/Sources/BreezCLI/StableBalance.swift
import Foundation
import BreezSdkSpark

// MARK: - Stable balance command names (for REPL completion)

let stableBalanceCommandNames: [String] = [
    "stable-balance get",
    "stable-balance set",
    "stable-balance unset",
]

// MARK: - Dispatch

func dispatchStableBalanceCommand(_ args: [String], sdk: BreezSdk) async {
    if args.isEmpty || args[0] == "help" {
        printStableBalanceHelp()
        return
    }

    let subName = args[0]
    let subArgs = Array(args.dropFirst())

    do {
        switch subName {
        case "get":
            try await handleStableBalanceGet(sdk, subArgs)
        case "set":
            try await handleStableBalanceSet(sdk, subArgs)
        case "unset":
            try await handleStableBalanceUnset(sdk, subArgs)
        default:
            print("Unknown stable-balance subcommand: \(subName). Use 'stable-balance help' for available commands.")
        }
    } catch {
        print("Error: \(error)")
    }
}

// MARK: - Help

private func printStableBalanceHelp() {
    print("\nStable balance subcommands:")
    print("  stable-balance \("get".padding(toLength: 22, withPad: " ", startingAt: 0))Get the stable balance active label")
    print("  stable-balance \("set".padding(toLength: 22, withPad: " ", startingAt: 0))Set the stable balance active label")
    print("  stable-balance \("unset".padding(toLength: 22, withPad: " ", startingAt: 0))Unset stable balance")
    print()
}

// MARK: - Handlers

// --- get ---

private func handleStableBalanceGet(_ sdk: BreezSdk, _ args: [String]) async throws {
    let settings = try await sdk.getUserSettings()
    printValue(settings.stableBalanceActiveLabel)
}

// --- set ---

private func handleStableBalanceSet(_ sdk: BreezSdk, _ args: [String]) async throws {
    guard !args.isEmpty else {
        print("Usage: stable-balance set <label>")
        return
    }

    try await sdk.updateUserSettings(request: UpdateUserSettingsRequest(
        sparkPrivateModeEnabled: nil,
        stableBalanceActiveLabel: .set(label: args[0])
    ))
    let settings = try await sdk.getUserSettings()
    printValue(settings)
}

// --- unset ---

private func handleStableBalanceUnset(_ sdk: BreezSdk, _ args: [String]) async throws {
    try await sdk.updateUserSettings(request: UpdateUserSettingsRequest(
        sparkPrivateModeEnabled: nil,
        stableBalanceActiveLabel: .unset
    ))
    let settings = try await sdk.getUserSettings()
    printValue(settings)
}

File: crates/breez-sdk/bindings/examples/cli/langs/kotlin-multiplatform/src/main/kotlin/StableBalance.kt
import breez_sdk_spark.*
import org.jline.reader.LineReader

data class StableBalanceCliCommand(
    val name: String,
    val description: String,
    val run: suspend (sdk: BreezSdk, reader: LineReader, args: List<String>) -> Unit
)

val STABLE_BALANCE_COMMAND_NAMES = listOf(
    "get",
    "set",
    "unset",
)

fun buildStableBalanceRegistry(): Map<String, StableBalanceCliCommand> {
    return mapOf(
        "get" to StableBalanceCliCommand("get", "Get the stable balance active label", ::handleStableBalanceGet),
        "set" to StableBalanceCliCommand("set", "Set the stable balance active label", ::handleStableBalanceSet),
        "unset" to StableBalanceCliCommand("unset", "Unset stable balance", ::handleStableBalanceUnset),
    )
}

suspend fun dispatchStableBalanceCommand(
    args: List<String>,
    sdk: BreezSdk,
    registry: Map<String, StableBalanceCliCommand>,
    reader: LineReader,
) {
    if (args.isEmpty() || args[0] == "help") {
        println()
        println("Stable balance subcommands:")
        registry.keys.sorted().forEach { name ->
            val cmd = registry[name]!!
            println("  stable-balance %-20s %s".format(name, cmd.description))
        }
        println()
        return
    }

    val subName = args[0]
    val subArgs = args.drop(1)

    val cmd = registry[subName]
    if (cmd == null) {
        println("Unknown stable-balance subcommand: $subName. Use 'stable-balance help' for available commands.")
        return
    }

    try {
        cmd.run(sdk, reader, subArgs)
    } catch (e: Exception) {
        println("Error: ${e.message}")
    }
}

// ---------------------------------------------------------------------------
// Stable balance command handlers
// ---------------------------------------------------------------------------

suspend fun handleStableBalanceGet(sdk: BreezSdk, reader: LineReader, args: List<String>) {
    val settings = sdk.getUserSettings()
    printValue(settings.stableBalanceActiveLabel)
}

suspend fun handleStableBalanceSet(sdk: BreezSdk, reader: LineReader, args: List<String>) {
    if (args.isEmpty()) {
        println("Usage: stable-balance set <label>")
        return
    }

    sdk.updateUserSettings(
        UpdateUserSettingsRequest(
            sparkPrivateModeEnabled = null,
            stableBalanceActiveLabel = StableBalanceActiveLabel.Set(label = args[0]),
        )
    )
    val settings = sdk.getUserSettings()
    printValue(settings)
}

suspend fun handleStableBalanceUnset(sdk: BreezSdk, reader: LineReader, args: List<String>) {
    sdk.updateUserSettings(
        UpdateUserSettingsRequest(
            sparkPrivateModeEnabled = null,
            stableBalanceActiveLabel = StableBalanceActiveLabel.Unset,
        )
    )
    val settings = sdk.getUserSettings()
    printValue(settings)
}
